### PR TITLE
Allow to set a truststore password in InternalCommunicationConfig

### PR DESCRIPTION
**Pull request by elukey**
Most of the java truststores need a password (if not all, I am not aware of the possibility to avoid setting one in recent JVM tools), meanwhile in Presto there seems to be no possibility to set one. This is really helpful when the CA cert is not contained in the keystore (for example, when one wants to use the system's default cacerts).

My use case: I have a cluster composed by one presto coordinator + 5 workers, with kerberos + internal https. The TLS certificates stored in the keystores are signed by a CA that it is included in the default list (`/etc/ssl/certs/java/cacerts` on Debian), but so far (up to 0.226) simply adding the following config to Presto's jvm settings made everything work:
```
-Djavax.net.ssl.trustStore=/etc/ssl/certs/java/cacerts
-Djavax.net.ssl.trustStorePassword=somepassword
``` 
After upgrading to 0.246 the above settings don't work anymore, I believe due to the explicit setting of the Truststore path happened in https://github.com/prestodb/presto/commit/6a7c4c81957cb4a85bc316bacf597c9fbe568f34#diff-474163b9662ed64fc2a0e14bfc2fb2a3d85517f387937192e4c0ce348850a502. Not sure why the password wasn't added, using a separate truststore is probably not a common use case and people didn't see the issue before.

PR: #15207

Test plan - (Please fill in how you tested your changes)
This should be a trivial change, already supported by Trino/PrestoSQL from a while (see https://github.com/trinodb/trino/pull/785/files).
I have a little presto testing environment, with 0.246 deployed using the release tarball downloaded from Maven Central. I checked out the prestodb repo up to the 0.246 git tag, applied this commit on top and finally built with `./mvnw clean install -DskipTests`. The copied `presto-main/target/presto-main-0.246.jar` to the testing environment and tested the new option. The https TLS authentication issues completely disappeared from the logs.

```
== RELEASE NOTES ==

General Changes
* New configuration parameter `internal-communication.https.trust-store-password` to set the Java Truststore password used for https in internal communications between nodes.

### CODE REVIEW

**Comment by elukey** [comment]
Really strange, only one failure in Travis over a lot of successes, but from the logs I can't really judge if it was caused from my change or not (it doesn't seem so but I am not that expert).

**Comment by elukey** [comment]
Any feedback on the patch? :)

File: presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationConfig.java [hunk 1]
```diff
@@ -84,13 +85,25 @@ public String getTrustStorePath()
         return trustStorePath;
     }
 
+    public String getTrustStorePassword()
```
Comment (aweisberg) [inline presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationConfig.java:88]: Can you move this getter to be next to the corresponding setter?

**Review by aweisberg — CHANGES_REQUESTED** [review]
If you looked at the TrinoDB PR for inspiration for this can you add "Co-authored-by: Chance Zibolski <czibolsk@redhat.com>" as the last line in your commit message (with one line of separation from everything above).

We try to be very careful to credit authors from other projects.

File: presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationConfig.java (hunk 1 above)
Comment (elukey) [reply presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationConfig.java:88]: Sure! Just updated my PR accordingly.

**Comment by elukey** [comment]
@aweisberg thanks for the review, just updated the commit msg and the code, hope it is good now :)

**Review by aweisberg — CHANGES_REQUESTED** [review]
I messed up advising you on the co-authored by. It needs to be `Co-authored-by: Chance Zibolski <czibolsk@redhat.com>` for GH to recognize it properly.

**Review by aweisberg — APPROVED** [review]
LGTM. Thanks!

**Comment by elukey** [comment]
@aweisberg can this be merged?

**Comment by ashishtadose** [comment]
Thank @elukey, this would help to configure trustore passwords beyond jvm opts. 
I did a review, looks good to me as well. 

Would be good to update trustStorePath & trustStorePassword in the doc of [secured internal communication](https://prestodb.io/docs/current/security/internal-communication.html)  may be in this PR or separate one.

**Pull request merged**